Extension/src/LanguageServer/settings.ts

```diff
@@ -691,18 +691,22 @@ function populateEditorConfig(rootUri: vscode.Uri | undefined, document: vscode.
     vscode.workspace.applyEdit(edits).then(() => vscode.window.showTextDocument(document));
 }
 
-export function generateEditorConfig(rootUri?: vscode.Uri): void {
+export async function generateEditorConfig(rootUri?: vscode.Uri): Promise<void> {
+    let document: vscode.TextDocument;
     if (rootUri !== undefined) {
         // If a folder is open and '.editorconfig' exists at the root, use that.
         const uri: vscode.Uri = vscode.Uri.joinPath(rootUri, ".editorconfig");
         const edits: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
         edits.createFile(uri, { ignoreIfExists: true, overwrite: false });
-        vscode.workspace.applyEdit(edits).then(() => {
-            vscode.workspace.openTextDocument(uri).then(
-                (document) => populateEditorConfig(rootUri, document),
-                () => vscode.workspace.openTextDocument().then((document) => populateEditorConfig(rootUri, document)));
-        }, () => vscode.workspace.openTextDocument().then((document) => populateEditorConfig(rootUri, document)));
+
+        try {
```

Comment:
To keep the intent of the previous code, this should be rewritten as:
```
try {
  await vscode.workspace.applyEdit(edits);
  document = await vscode.workspace.openTextDocument(uri);
} catch {
  document = await vscode.workspace.openTextDocument();  // no uri...
}
populateEditorConfig(document);
```

Reply:
please take a look at the modified code.